src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -272,120 +308,167 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
+                                                    &self.metrics,
+                                                    truncation_offset,
+                                                    &mut buffer,
+                                                )
+                                                .await?;
+                                            seq_id = FIRST_SEQ_ID;
+                                            let update_vnode_bitmap = barrier
+                                                .as_update_vnode_bitmap(self.actor_context.id);
+                                            let barrier_epoch = barrier.epoch;
+                                            yield Message::Barrier(barrier);
+                                            write_state_post_write_barrier
+                                                .post_yield_barrier(update_vnode_bitmap.clone())
+                                                .await?;
+                                            if let Some(vnode_bitmap) = update_vnode_bitmap {
+                                                // Apply Vnode Update
+                                                read_state.update_vnode_bitmap(vnode_bitmap);
+                                                initial_write_epoch = barrier_epoch;
+                                                input = stream;
+                                                initial_write_state = write_state;
+                                                continue 'recreate_consume_stream;
+                                            } else {
+                                                write_future_state =
+                                                    WriteFuture::receive_from_upstream(
+                                                        stream,
+                                                        write_state,
+                                                    );
+                                            }
                                         }
                                     }
                                     Message::Chunk(chunk) => {
-                                        let start_seq_id = seq_id;
-                                        seq_id += chunk.cardinality() as SeqIdType;
-                                        let end_seq_id = seq_id - 1;
-                                        let epoch = write_state.epoch().curr;
-                                        if let Some(chunk_to_flush) = buffer.add_or_flush_chunk(
-                                            start_seq_id,
-                                            end_seq_id,
-                                            chunk,
-                                            epoch,
-                                        ) {
-                                            write_future = WriteFuture::flush_chunk(
+                                        if clean_state && buffer.size() + chunk.cardinality()
```

Comment:
Should be
```
if let Some(chunk) = buffer.add_or_flush_chunk(...) {
    not update seq_id, but only get start_seq_id and end_seq_id
    if clean_state {
        write_future_state = pause(chunk);
    } else {
        write_future_state = flush_chunk(chunk);
        clean_state = false;
        update seq_id
    }
}
```

Reply:
Fixed, `add_or_flush_chunk` does not update any state, unless it actually flushes a chunk, so it's safe to call directly, rather than check size before.